Mathlib/CategoryTheory/Monoidal/Functor.lean

```diff
@@ -192,8 +192,78 @@ theorem LaxMonoidalFunctor.associativity_inv (F : LaxMonoidalFunctor C D) (X Y Z
 
 end
 
+/-- A oplax monoidal functor is a functor `F : C ⥤ D` between monoidal categories,
+equipped with morphisms `ε : F.obj (𝟙_ C) ⟶ 𝟙 _D` and `μ X Y : F.obj (X ⊗ Y) ⟶ F.obj X ⊗ F.obj Y`,
+satisfying the appropriate coherences. -/
+structure OplaxMonoidalFunctor extends C ⥤ D where
+  /-- unit morphism -/
+  ε : obj (𝟙_ C) ⟶ 𝟙_ D
```

Comment:
I don't think we should use the same letter as we do for `LaxMonoidal`. Eg `Monad` and `Comonad` use different letters for the (co)unit and (co)multiplication and the nlab page for oplax monoidal things uses a different letter than `\mu` for the cotensorator. It also seems like it would make it hard to talk about functors which are both lax and oplax (strong monoidal functors are the most common of these but there are others, like how we have both hopf and frobenius bialgebras). I would prefer if we use `η` and `δ`

Reply:
Excellent point! I'll rename as you suggested.

Reply:
Awesome! Other than that everything looks good. Also, if you're going to be working on this stuff, I had a draft PR #10845 which in particular added oplax monoidal functors (called colax monoidal) and might be useful to compare with. Had to drop it because of other priorities and a bug in lean but maybe I'll revive it sometime?